backupdb/management/commands/backupdb.py

```diff
@@ -51,48 +52,54 @@ class Command(BaseBackupDbCommand):
     def handle(self, *args, **options):
         super(Command, self).handle(*args, **options)
 
-        from django.conf import settings
+        current_time_string = timezone.now().isoformat()
+
+        backup_suffix = options['backup_name']
+        if backup_suffix is None:
+            backup_suffix = current_time_string
 
-        current_time = time.strftime('%F-%s')
-        backup_name = options['backup_name'] or current_time
         show_output = options['show_output']
 
+        backup_directory = get_backup_directory()
+
         # Ensure backup dir present
-        if not os.path.exists(BACKUP_DIR):
-            os.makedirs(BACKUP_DIR)
+        if not os.path.exists(backup_directory):
+            os.makedirs(backup_directory)
 
         # Loop through databases
         for db_name, db_config in settings.DATABASES.items():
-            with section("Backing up '{0}'...".format(db_name)):
-                # Get backup config for this engine type
-                engine = db_config['ENGINE']
-                backup_config = BACKUP_CONFIG.get(engine)
-                if not backup_config:
-                    raise SectionWarning("Backup for '{0}' engine not implemented".format(engine))
-
-                # Get backup file name
-                backup_base_name = '{db_name}-{backup_name}.{backup_extension}.gz'.format(
-                    db_name=db_name,
-                    backup_name=backup_name,
-                    backup_extension=backup_config['backup_extension'],
-                )
-                backup_file = os.path.join(BACKUP_DIR, backup_base_name)
-
-                # Find backup command and get kwargs
-                backup_func = backup_config['backup_func']
-                backup_kwargs = {
-                    'backup_file': backup_file,
-                    'db_config': db_config,
-                    'show_output': show_output,
-                }
-                if backup_func is do_postgresql_backup:
-                    backup_kwargs['pg_dump_options'] = options['pg_dump_options']
-
-                # Run backup command
-                try:
-                    backup_func(**backup_kwargs)
-                    logger.info("Backup of '{db_name}' saved in '{backup_file}'".format(
+            logger.info("Starting to backup {!r}".format(db_name))
+            engine = db_config['ENGINE']
+            try:
+                backup_cls = BACKUP_CONFIG[engine]
+            except KeyError:
+                logger.error(
+                    "Backup for {!r} engine not implemented".format(engine))
+                continue  # TODO: Raise an error by default
+                          # TODO: implement --ignore-errors
+
+            fname = '{db}-{suffix}.{ext}.gz'.format(
+                db=db_name,
+                suffix=backup_suffix,
+                ext=backup_cls.extension,
+            )
+            absolute_fname = os.path.join(backup_directory, fname)
```

Comment:
No, you're totally right. I'm just not sure where to get the absolute path. Should it be `os.path.join(PROJECT_PATH, BACKUP_DIR)`, because `BACKUP_DIRECTORY` doesn't have to be an absolute directory.